datastore/src/main/scala/org/allenai/datastore/Datastore.scala

```diff
@@ -382,6 +384,31 @@ class Datastore(val name: String, val s3: AmazonS3Client) extends Logging {
   // Putting data into the datastore
   //
 
+  private def multipartUpload(path: Path, locator: Locator): Unit = {
+    val tm = new TransferManager(s3)
+    try {
+      val request = new PutObjectRequest(bucketName, locator.s3key, path.toFile)
+      request.setGeneralProgressListener(new ProgressListener {
+        private var lastLogMessage = System.currentTimeMillis()
```

Comment:
Should you initialize to `System.currentTimeMillis() - 1000`, so that you get an initial log message right away?